### PR TITLE
Disallow capital letter at the beginning in CamelCasePropertyName rule

### LINKED ISSUE #276: CamelCasePropertyName does allow Capitals at the beginning while CamelCaseVariableName does not
The rule CamelCasePropertyName does allow capitals at the beginning while CamelCaseVariableName does not.

https://github.com/phpmd/phpmd/blob/2.2.1/src/main/php/PHPMD/Rule/Controversial/CamelCasePropertyName.php#L70
https://github.com/phpmd/phpmd/blob/2.2.1/src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php#L93

@manuelpichler do you know of a particular reason why this is so?

### PR DESCRIPTION
**Pull request by martzd**
Type: bugfix 
Issue: Resolves #276
Breaking change: yes, capital letter at the beggining of property name is not allowed anymore in the CamelCasePropertyName rule

### CODE REVIEW

**Review by kylekatarnls — APPROVED** [review]

File: src/test/resources/files/Rule/Controversial/CamelCasePropertyName/testRuleDoesApplyForPropertyNameWithCapital.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * @link http://phpmd.org/
  */
 
-class testRuleDoesNotApplyForPropertyNameWithCapital
+class testRuleDoesApplyForPropertyNameWithCapital
 {
     public $AlsoValidPropertyName;
```
Comment (kylekatarnls) [inline src/test/resources/files/Rule/Controversial/CamelCasePropertyName/testRuleDoesApplyForPropertyNameWithCapital.php:20]:
```suggestion
    public $NotValidPropertyName;
```
Would be less missleading.

File: src/test/php/PHPMD/Rule/Controversial/CamelCasePropertyNameTest.php [hunk 2]
```diff
@@ -42,15 +42,15 @@ public function testRuleDoesNotApplyForValidPropertyName()
     }
 
     /**
-     * Tests that the rule does NOT apply for an property name
+     * Tests that the rule does apply for an property name
```
Comment (kylekatarnls) [inline src/test/php/PHPMD/Rule/Controversial/CamelCasePropertyNameTest.php:45]:
```suggestion
     * Tests that the rule does apply for a property name
```
While modifying this line, we could take the opportunity to fix this typo.

Comment (martzd) [reply src/test/php/PHPMD/Rule/Controversial/CamelCasePropertyNameTest.php]: Fixed.

File: src/test/resources/files/Rule/Controversial/CamelCasePropertyName/testRuleDoesApplyForPropertyNameWithCapital.php (hunk 1 above)
Comment (martzd) [reply src/test/resources/files/Rule/Controversial/CamelCasePropertyName/testRuleDoesApplyForPropertyNameWithCapital.php]: Fixed.

**Review by kylekatarnls — APPROVED** [review]

**Review by MarkVaughn — APPROVED** [review]

**Comment by tvbeek** [comment]
I miss read the BC part and pressed merge to fast, so I created a revert PR. Sorry for that.

**Pull request merged**